examples/misc/lib/language_tour/patterns/algebraic_datatypes.dart

```diff
@@ -0,0 +1,20 @@
+import 'dart:math' as math;
+
+// #docregion algebraic_datatypes
+sealed class Shape {}
+
+class Square implements Shape {
+  final double length;
+  Square(this.length);
+}
+
+class Circle implements Shape {
+  final double radius;
+  Circle(this.radius);
+}
+
+double calculateArea(Shape shape) => switch (shape) {
+      Square(length: var l) => l * l,
+      Circle(radius: var r) => math.pi * r * r
+    };
+// #enddocregion algebraic_datatypes
```

Comment:
Did you get a bunch of errors on this? I've been trying to implement the same example for the Branches page and it's giving me all kinds of errors...

Reply:
I needed to import the math library to get it working:

```
import 'dart:math' as math;
```